File: source_code/project4_spring/chat/src/main/java/com/service/chat/dto/CustomResult.java
package com.service.chat.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class CustomResult {
    private int status;
    private String message;
    private Object data;

}

File: source_code/project4_flutter/lib/features/property_detail/models/custom_result.dart
import 'package:json_annotation/json_annotation.dart';

part 'custom_result.g.dart';

@JsonSerializable()
class CustomResult {
  int status;
  String message;
  dynamic data;

  CustomResult(
      {required this.status, required this.message, required this.data});

  factory CustomResult.fromJson(Map<String, dynamic> json) =>
      _$CustomResultFromJson(json);

  Map<String, dynamic> toJson() => _$CustomResultToJson(this);
}
